### PR TITLE
Add `bun` converter

### PR DESCRIPTION
**Pull request by colinhacks**
- Adds npm-to-bun conversion

Context: The `bun` package manager is also runtime-agnostic and can be used in Node.js or Bun projects. Bun users primarily use `bun` for package management. https://bun.sh/package-manager

### CODE REVIEW

File: test/index.spec.ts [hunk 1]
```diff
@@ -5,188 +5,318 @@ import convert from '../src'
...
     [
       'npm pack --pack-destination=foobar',
       'yarn pack --filename foobar',
-      'pnpm pack --pack-destination foobar'
-    ]
-  ]
+      'pnpm pack --pack-destination foobar',
+      "npm pack --pack-destination foobar\n# couldn't auto-convert command",
+    ],
+  ] as [string, string, string, string][]
```
Comment (armano2) [inline test/index.spec.ts]: can you move it up to

```ts
const tests: [npm: string, yarn: string, pnpm: string, bun: string][] = [
```

File: dist/npm-to-yarn.mjs [hunk 2]
```diff
@@ -485,6 +485,133 @@ function npmToPnpm(_m, command) {
...
+        case 'remove':
+        case 'r':
+        case 'rm':
+            args[0] = 'remove';
+            args = convertInstallArgs(args);
+            break;
+        case 'cache':
+            if (args[1] === 'clean') {
+                args = ['pm', 'cache', 'rm'];
+                args = args.concat(args.slice(2));
```
Comment (armano2) [inline dist/npm-to-yarn.mjs:549]: this line seem to be erroring

Comment (armano2) [inline dist/npm-to-yarn.mjs]: this is not correct as rm is going to be present 2 times

i'm not sure about syntax, but maybe soemthign like this?
```suggestion
                args = ['pm', 'cache', 'rm'].concat(args.slice(2));
```

or

```suggestion
                args = args.splice(0, 2, 'pm', 'cache', 'rm');
```

Comment (colinhacks) [reply dist/npm-to-yarn.mjs]: Good catch

**Comment by bgub** [comment]
@armano2 do you think this is ready to merge?

**Comment by bgub** [comment]
@colinhacks I'll merge this now. Thanks so much for the PR!

**Comment by bgub** [comment]
@colinhacks released in version 2.1.0 :)

**Pull request merged**